react/src/components/ServiceLauncherModal.tsx

```diff
@@ -115,11 +208,38 @@ const ServiceLauncherModal: React.FC<ServiceLauncherProps> = ({
           },
         },
       };
-      if (resourceSlots?.['cuda.shares']) {
-        body['config'].resources['cuda.shares'] = values.gpu;
-      }
-      if (resourceSlots?.['cuda.device']) {
-        body['config'].resources['cuda.device'] = values.gpu;
+      // Set AI accelerator value if set
+      // Currently, we only support one AI accelerator per session
+      if (values.resource.acceleratorType) {
+        const acceleratorTypeName: string =
+          values.resource.acceleratorType?.value;
+        // FIXME: temporally add switch-case
+        switch (acceleratorTypeName) {
+          case 'cuda.shares':
+            body['config'].resources['cuda.shares'] =
+              values.resource.accelerator;
+            break;
+          case 'cuda.device':
+            body['config'].resources['cuda.device'] =
+              values.resource.accelerator;
+            break;
+          case 'rocm.device':
+            body['config'].resources['rocm.device'] =
+              values.resource.accelerator;
+            break;
+          case 'tpu.device':
+            body['config'].resources['tpu.device'] =
+              values.resource.accelerator;
+            break;
+          case 'ipu.device':
+            body['config'].resources['ipu.device'] =
+              values.resource.accelerator;
+            break;
+          case 'warboy.device':
+            body['config'].resources['warboy.device'] =
+              values.resource.accelerator;
+            break;
+        }
       }
```

Comment:
We can make it shorter.

```suggestion
      if (values.resource.acceleratorType) {
        const acceleratorTypeName: string = values.resource.acceleratorType?.value;
        body['config'].resources[acceleratorTypeName] = values.resource.accelerator;
      }
```